raft/storage.go

```diff
@@ -248,7 +248,7 @@ func (ms *MemoryStorage) Append(entries []pb.Entry) error {
 	offset := entries[0].Index - ms.ents[0].Index
 	switch {
 	case uint64(len(ms.ents)) > offset:
-		ms.ents = append([]pb.Entry{}, ms.ents[:offset]...)
+		ms.ents = ms.ents[:offset]
```

Comment:
the reason is that all entries in ms.ents should be readonly. we might return these entries to users. we shall never modify any existing entries.